rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2232,15 +2231,86 @@ void getProposedFederationAddress_whenStorageProviderReturnsProposedFederation_s
         assertEquals(proposedFederation.getAddress(), actualProposedFederationAddress.get());
     }
 
+    @Test
+    void getProposedFederatorPublicKeyOfType_whenFederationIsEmpty_shouldReturnEmpty() {
+        // Act
+        Optional<byte[]> actualPublicKey = federationSupport.getProposedFederatorPublicKeyOfType(0, FederationMember.KeyType.BTC);
+
+        // Assert
+        assertFalse(actualPublicKey.isPresent());
+    }
+
+    @Test
+    void getProposedFederatorPublicKeyOfType_whenStorageProviderReturnsProposedFederationAndKeyTypeIsBTC_shouldReturnPublicKey() {
+        // Arrange
+        List<BtcECKey> federationKeys = BitcoinTestUtils.getBtcEcKeysFromSeeds(
+            new String[] { "fa01", "fa02", "fa03", "fa04", "fa05", "fa06", "fa07", "fa08", "fa09" }, true);
+        Federation proposedFederation = P2shErpFederationBuilder.builder().withMembersBtcPublicKeys(federationKeys).build();
+        FederationMember.KeyType keyType = FederationMember.KeyType.BTC;
+        byte[] expectedPublicKey = proposedFederation.getMembers().get(0).getPublicKey(keyType).getPubKey(true);
```

Comment:
```suggestion
        BtcECKey publicKey1 = BtcECKey.fromPublicOnly(Hex.decode("0261631efa5f32602a3e2b1dc89d648c3212f17fd87e26614cb3a1d9516346098a");
        List<BtcECKey> federationKeys = Arrays.asList(
            publicKey1,
            BtcECKey.fromPublicOnly(Hex.decode(""),
            etc
        );
        Federation proposedFederation = P2shErpFederationBuilder.builder().withMembersBtcPublicKeys(federationKeys).build();

        FederationMember.KeyType keyType = FederationMember.KeyType.BTC;
        byte[] expectedPublicKey = publicKey1.getPubKey(true);
```

wdyt?

Reply:
For me it is as explicit as it should be. Open to more thoughts/ideas.